### DISCUSSION TITLE
Ceph OSD encrypt module

### DISCUSSION THREAD

**Discussion by Jiawei0227** [rook/rook#9119] [Q&A] [answered]
Hey guys, I am trying to understand more about ceph security implementation inside rook.

Especially I am interested to learn if containerized components(like Ceph OSD pods) call encryption libraries within the base image of itself or does it delegate to a host process that can link with FIPS-certified modules to do encryptions.

Appreciate anyone who can provide any guidance or helps, links here. Thanks so much!

**Comment by leseb** [comment, marked as answer]
The OSD encryption capability is using LUKS. With the tooling (`cryptsetup`) coming from the base image. The rest is handled by the Kernel. If FIPS is enabled on the host, the OSD encryption will inherit from it. Does that answer your question?

**Reply by Jiawei0227** [reply]
Thanks for the reply! So just to confirm my understanding here, there will be two parts involved in OSD Encryption using LUKS:
- cryptsetup binary
- dm-crypt kernel module

And in the case of containerized OSD image, the binary will be from the base image. And the dm-crypt module is from the host kernel. So if host has FIPS-certified kernel module, it will use it.

**Reply by leseb** [reply]
Yes, that's all correct.
